doc/directives.md

```diff
@@ -0,0 +1,117 @@
+Dartdoc supports several **directives** within Dart doc comments. Each directive
+is then processed during documentation generation, and new text is inserted in
+place of the directive. Not all directives are supported for package
+documentation at https://pub.dev/.
+
+A doc comment directive consists of either one directive tag, which looks
+something like `{@DIRECTIVE ARG=VALUE ...}`, or two such tags (an opening and a
+closing tag), with content in between.
+
+The supported directives are listed below:
+
+## `@nodoc` - Do not include documentation
+
+An element whose doc comment should not appear in the generated documenation can
+include the `@nodoc` directive.
+
+## `{@category}` and `{@subCategory}` - Categories
+
+Elements such as libraries and classes can be grouped into categories and
+sub-categories by adding `{@category CATEGORY NAME}` and
+`{@subCategory SUB-CATEGORY NAME}` in doc comments. Each category then gets its
+own documentation page, listing all of the categorized elements.
+
+## `{@template}` and `{@macro}` - Templates and macros
+
+TODO(srawlins): Document
+
+## `{@example}` - Examples (deprecated)
+
+Examples from the file system can be inlined by using the `{@example}`
+directive. The file path, the region, and the example language can all be
+specified with the following syntax:
+
+```none
+{@example PATH [region=NAME] [lang=NAME]}
+```
+
+All example file names must have the extension, `.md`, and this extension must
+not be specified in the example `PATH`. `PATH` must be specified as a relative
+path from the root of the project for which documentation is being generated.
+Given `dir/file.dart` as `PATH`, an example will be extracted from
+`dir/file.dart.md`, relative to the project root directory.
+
+During doc generation, dartdoc will replace the `{@example}` directive with the
+contents of the example file, verbatim.
+
+TODO(srawlins): Document region, lang, `--example-path-prefix`.
+
+## `{@inject-html}` - Injected HTML
+
+HTML can be rendered unmodified by including it between `{@inject-html}` and
+`{@end-inject-html}` directive tags. The tags take no arguments:
+
+```none
+{@inject-html}
+INJECTED HTML
+{@end-inject-html}
```

Comment:
Do we want to make the examples valid code by inserting `\\\` before each line? It's not necessary, but it might make it easier to copy/paste.